fud2/fud-core/src/run.rs

```diff
@@ -87,6 +87,77 @@ impl EmitBuild for EmitBuildFn {
     }
 }
 
+/// A config variable.
+#[derive(Debug, Clone)]
+pub enum ConfigVar {
+    /// The key for the config variable.
+    Var(String),
+    /// The key for the config variable followed by the value it should be if the key is not found.
+    VarOr(String, String),
+}
+
+/// The data required to emit a single op.
+pub struct OpEmitData {
+    /// The name of the op.
```

Comment:
Let's refer to the Ninja this will generate. This isn't actually the name of the op—that's stored elsewhere. Here, this is the name of the Ninja rule we'll emit.